asv_bench/benchmarks/frame_ctor.py

```diff
@@ -0,0 +1,1706 @@
+        self.d = dict([(col, self.df[col]) for col in self.df.columns])
+
+    def time_frame_ctor_dtindex_FY5253x2__variation_nearest(self):
+        DataFrame(self.d)
+
+
+class frame_ctor_dtindex_Hourx1(object):
+    goal_time = 0.2
+
+    def setup(self):
+
+        def get_period_count(start_date, off):
+            self.ten_offsets_in_days = ((start_date + (off * 10)) - start_date).days
+            if (self.ten_offsets_in_days == 0):
+                return 1000
+            else:
+                return min((9 * ((Timestamp.max - start_date).days // self.ten_offsets_in_days)), 1000)
+
+        def get_index_for_offset(off):
+            self.start_date = Timestamp('1/1/1900')
+            return date_range(self.start_date, periods=min(1000, get_period_count(self.start_date, off)), freq=off)
+        self.idx = get_index_for_offset(Hour(1, **{}))
+        self.df = DataFrame(np.random.randn(len(self.idx), 10), index=self.idx)
+        self.d = dict([(col, self.df[col]) for col in self.df.columns])
+
+    def time_frame_ctor_dtindex_Hourx1(self):
+        DataFrame(self.d)
+
+
+class frame_ctor_dtindex_Hourx2(object):
+    goal_time = 0.2
+
+    def setup(self):
+
+        def get_period_count(start_date, off):
+            self.ten_offsets_in_days = ((start_date + (off * 10)) - start_date).days
+            if (self.ten_offsets_in_days == 0):
+                return 1000
+            else:
+                return min((9 * ((Timestamp.max - start_date).days // self.ten_offsets_in_days)), 1000)
+
+        def get_index_for_offset(off):
+            self.start_date = Timestamp('1/1/1900')
+            return date_range(self.start_date, periods=min(1000, get_period_count(self.start_date, off)), freq=off)
+        self.idx = get_index_for_offset(Hour(2, **{}))
+        self.df = DataFrame(np.random.randn(len(self.idx), 10), index=self.idx)
+        self.d = dict([(col, self.df[col]) for col in self.df.columns])
+
+    def time_frame_ctor_dtindex_Hourx2(self):
+        DataFrame(self.d)
+
+
+class frame_ctor_dtindex_LastWeekOfMonthx1(object):
+    goal_time = 0.2
+
+    def setup(self):
+
+        def get_period_count(start_date, off):
+            self.ten_offsets_in_days = ((start_date + (off * 10)) - start_date).days
+            if (self.ten_offsets_in_days == 0):
+                return 1000
+            else:
+                return min((9 * ((Timestamp.max - start_date).days // self.ten_offsets_in_days)), 1000)
+
+        def get_index_for_offset(off):
+            self.start_date = Timestamp('1/1/1900')
+            return date_range(self.start_date, periods=min(1000, get_period_count(self.start_date, off)), freq=off)
+        self.idx = get_index_for_offset(LastWeekOfMonth(1, **{'week': 1, 'weekday': 1, }))
+        self.df = DataFrame(np.random.randn(len(self.idx), 10), index=self.idx)
+        self.d = dict([(col, self.df[col]) for col in self.df.columns])
+
```

Comment:
I don't think we should try to auto-generate them, also not after cleaning it up manually.

- Use this auto-converted and long code for now as the initial start (the other things like being able to compare against commit, update the perf_test scripts etc are more important initially)
- Afterwards we can clean this up and deduplicate code, but then I would just use standard subclassing/reusing of functions etc to achieve that instead of some auto-generation of classes.